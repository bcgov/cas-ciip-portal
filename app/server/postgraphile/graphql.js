const {
  createPostGraphileSchema,
  withPostGraphileContext,
} = require("postgraphile");
const { graphql } = require("graphql");
const authenticationPgSettings = require("./authenticationPgSettings");
const { pgPool, getDatabaseUrl } = require("../db/databaseConnectionService");
const postgraphileOptions = require("./postgraphileOptions");

const pgSettings = (req) => {
  const opts = {
    ...authenticationPgSettings(req),
  };
  return opts;
};

let postgraphileSchemaSingleton;

const postgraphileSchema = async () => {
  if (!postgraphileSchemaSingleton) {
    postgraphileSchemaSingleton = await createPostGraphileSchema(
      getDatabaseUrl(),
      process.env.DATABASE_SCHEMA || "ggircs_portal",
      postgraphileOptions()
    );
  }

  return postgraphileSchemaSingleton;
};

async function performQuery(query, variables, request) {
  const settings = pgSettings(request);

  return withPostGraphileContext(
    {
      pgPool,
      pgSettings: settings,
    },
    async (context) => {
      // `context` object, which should NOT be used outside of this
      // function.
      const result = graphql(
        await postgraphileSchema(),
        query,
        null,
        { ...context }, // You can add more to context if you like
        variables
      );
      return result;
    }
  );
}

module.exports = {
  performQuery,
};
