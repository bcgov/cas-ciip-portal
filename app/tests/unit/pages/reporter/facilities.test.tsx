import React from 'react';
import FacilitiesList from 'pages/reporter/facilities';
import {shallow} from 'enzyme';
import {facilitiesQueryResponse} from '__generated__/facilitiesQuery.graphql';

describe('The reporter facilities list page', () => {
  it('matches the snapshot', () => {
    const query: facilitiesQueryResponse['query'] = {
      session: {' $fragmentRefs': {defaultLayout_session: true}},
      ' $fragmentRefs': {FacilitiesListContainer_query: true}
    };

    const pageUnderTest = shallow(
      <FacilitiesList router={null} query={query} />
    );
    expect(pageUnderTest).toMatchSnapshot();
  });
});
