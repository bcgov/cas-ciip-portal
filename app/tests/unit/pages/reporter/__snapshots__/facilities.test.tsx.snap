// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The reporter facilities list page matches the snapshot 1`] = `
<Relay(DefaultLayout)
  session={
    Object {
      " $fragmentRefs": Object {
        "defaultLayout_session": true,
      },
    }
  }
  showSubheader={true}
  title="Facilities"
  width="wide"
>
  <Row
    className="justify-content-md-center mb-5"
    noGutters={false}
  >
    <Col>
      <ProgressStepIndicator
        steps={
          Array [
            Object {
              "badgeStyle": "light",
              "description": "CIIP Application not started",
              "number": 1,
            },
            Object {
              "badgeStyle": "warning",
              "description": "Application in progress",
              "number": 2,
            },
            Object {
              "badgeStyle": "info",
              "description": "Application submitted and in review",
              "number": 3,
            },
            Object {
              "badgeStyle": "success",
              "description": "Application approved or rejected",
              "number": 4,
            },
          ]
        }
        title="Steps to Apply for an Operator's Facility"
      />
    </Col>
  </Row>
  <Relay(FacilitiesList)
    query={
      Object {
        " $fragmentRefs": Object {
          "FacilitiesListContainer_query": true,
        },
        "session": Object {
          " $fragmentRefs": Object {
            "defaultLayout_session": true,
          },
        },
      }
    }
  />
</Relay(DefaultLayout)>
`;
